scripts/ota_updater/ota_updater.py

```diff
@@ -1 +1,153 @@
+            elif status == 403: # forbidden
+                print("Device ota disabled, aborting...")
+                client.disconnect()
+
+    elif msg.topic.endswith('$fw/checksum'):
+        checksum = msg.payload
+
+        if userdata.get("published"):
+            if checksum == userdata.get('md5'):
+                print("Device back online. Update Successful!")
+            else:
+                print("Expecting checksum {}, got {}, update failed!".format(userdata.get('md5'), checksum))
+            client.disconnect()
+        else:
+            if checksum != userdata.get('md5'): # save old md5 for comparison with new firmware
+                userdata.update({'old_md5': checksum})
+            else:
+                print("Device firmware already up to date with md5 checksum: {}".format(checksum))
+                client.disconnect()
+
+    elif msg.topic.endswith('ota/enabled'):
+        if msg.payload == 'true':
+            userdata.update({'ota_enabled': True})
+        else:
+            print("Device ota disabled, aborting...")
+            client.disconnect()
+
+    if ( not userdata.get("published") ) and ( userdata.get('ota_enabled') ) and \
+       ( 'old_md5' in userdata.keys() ) and ( userdata.get('md5') != userdata.get('old_md5') ):
+        # push the firmware binary
+        userdata.update({"published": True})
+        topic = "{base_topic}{device_id}/$implementation/ota/firmware/{md5}".format(**userdata)
+        print("Publishing new firmware with checksum {}".format(userdata.get('md5')))
+        client.publish(topic, userdata['firmware'])
+
+
+def main(broker_host, broker_port, broker_username, broker_password, base_topic, device_id, firmware):
+    # initialise mqtt client and register callbacks
+    client = mqtt.Client()
+    client.on_connect = on_connect
+    client.on_message = on_message
+
+    # set username and password if given
+    if broker_username and broker_password:
+        client.username_pw_set(broker_username, broker_password)
+
+    # save data to be used in the callbacks
+    client.user_data_set({
+            "base_topic": base_topic,
+            "device_id": device_id,
+            "firmware": firmware
+        })
+
+    # start connection
+    print("Connecting to mqtt broker {} on port {}".format(broker_host, broker_port))
+    client.connect(broker_host, broker_port, 60)
+
+    # Blocking call that processes network traffic, dispatches callbacks and handles reconnecting.
+    client.loop_forever()
+
+
+if __name__ == '__main__':
+    import argparse
+
+    parser = argparse.ArgumentParser(
+        description='ota firmware update scirpt for ESP8226 implemenation of the Homie mqtt IoT convention.')
+
+    # ensure base topic always ends with a '/'
+    def base_topic_arg(s):
+        s = str(s)
+        if not s.endswith('/'):
+            s = s + '/'
+        return s
+
+    # specify arguments
+    parser.add_argument('-l', '--broker-host',     type=str,            required=True,
```

Comment:
Can you make both the broker host and port optional? A `127.0.0.1:1883` default seems like a good idea for local testing.